### PR TITLE
[core] Fix sprite sheet merging in Style::Impl::onSpriteLoaded

### PR DESCRIPTION
**Pull request by alexshalamov**
If we get a new sprite sheet from the server, we need to merge current sprite sheet with a new one, while overwriting duplicates and keeping old unique images in a style.

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] apply `needs changelog` label if a changelog is needed (remove label when added)

### CODE REVIEW

File: src/mbgl/style/style_impl.cpp [hunk 1]
```diff
@@ -341,9 +341,24 @@ void Style::Impl::onSourceDescriptionChanged(Source& source) {
 
 void Style::Impl::onSpriteLoaded(std::vector<Immutable<style::Image::Impl>> images_) {
     auto newImages = makeMutable<ImageImpls>(*images);
+    const auto cmp = [](const auto& a, const auto& b) { return a->id < b->id; };
+    assert(std::is_sorted(newImages->begin(), newImages->end(), cmp));
+
+    for (auto it = images_.begin(); it != images_.end();) {
+        const auto& image = *it;
+        auto first = std::lower_bound(newImages->begin(), newImages->end(), image, cmp);
```
Comment (mr1sunshine) [inline src/mbgl/style/style_impl.cpp]: nit: this could also be `const`

File: src/mbgl/style/style_impl.cpp [hunk 2]
```diff
@@ -341,9 +341,24 @@ void Style::Impl::onSourceDescriptionChanged(Source& source) {
 
 void Style::Impl::onSpriteLoaded(std::vector<Immutable<style::Image::Impl>> images_) {
     auto newImages = makeMutable<ImageImpls>(*images);
+    const auto cmp = [](const auto& a, const auto& b) { return a->id < b->id; };
```
Comment (mr1sunshine) [inline src/mbgl/style/style_impl.cpp]: I'm just thinking if this comparator could be added as friend operator (< and ==) for `style::Image::Impl`, so we could save some lines of code here and in `sprite_parser.cpp` as well.

Comment (alexshalamov) [reply src/mbgl/style/style_impl.cpp]: done, added operator <

File: src/mbgl/style/style_impl.cpp (hunk 1 above)
Comment (alexshalamov) [reply src/mbgl/style/style_impl.cpp]: :+1:

**Pull request merged**